orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -869,7 +872,36 @@ class SqlExecutionRepository(
 
       // Restore original object state.
       execution.stages.addAll(stages)
+      pipelineTrigger?.let {
+        execution.trigger = it
+      }
+    }
+  }
+
+  /**
+   * Converts a [PipelineTrigger] into a [PipelineRefTrigger] for storage.
```

Comment:
I believe this only handling state of pipeline executions. I wonder if this is the best place for this. I dont want to overload the class more than needed, and have it be focused for its one particular job.